### ISSUE TITLE
libnss3.so not found on SLES 15 SP4

### ISSUE THREAD

**Issue by D042657** [cypress-io/cypress#26480]
### Current behavior

I installed Cypress 12.9.0 on SLES 15 SP4 and then ran npx cypress open. I received the following error:

Now using node v16.20.0 (npm v8.19.4)
It looks like this is your first time using Cypress: 12.9.0

Cypress failed to start.

This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies

Please refer to the error below for more details.

----------

/home/d042657/.cache/Cypress/12.9.0/Cypress/Cypress: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory

----------

Platform: linux-x64 (Raspbian - 15.4)
Cypress Version: 12.9.0

### Desired behavior

npx cypress open starts the interactive GUI for cypress

### Test code to reproduce

npx cypress open

### Cypress Version

12.9.0

### Node version

v16.20

### Operating System

SLES 15 SP4

### Debug Logs

```shell
╰─⠠⠵ DEBUG=cypress:* cypress run
  cypress:cli:cli cli starts with arguments ["/home/XXXXXX/.nvm/versions/node/v16.20.0/bin/node","/home/XXXXXX/.nvm/versions/node/v19.9.0/bin/cypress","run"] +0ms
  cypress:cli NODE_OPTIONS is not set +0ms
  cypress:cli:cli program parsing arguments +2ms
  cypress:cli:cli running Cypress with args [ Command { _events: [Object: null prototype] { 'option:auto-cancel-after-failures': [Function (anonymous)], 'option:browser': [Function (anonymous)], 'option:ci-build-id': [Function (anonymous)], 'option:component': [Function (anonymous)], 'option:config': [Function (anonymous)], 'option:config-file': [Function (anonymous)], 'option:e2e': [Function (anonymous)], 'option:env': [Function (anonymous)], 'option:group': [Function (anonymous)], 'option:key': [Function (anonymous)], 'option:headed': [Function (anonymous)], 'option:headless': [Function (anonymous)], 'option:no-exit': [Function (anonymous)], 'option:parallel': [Function (anonymous)], 'option:port': [Function (anonymous)], 'option:project': [Function (anonymous)], 'option:quiet': [Function (anonymous)], 'option:record': [Function (anonymous)], 'option:reporter': [Function (anonymous)], 'option:reporter-options': [Function (anonymous)], 'option:spec': [Function (anonymous)], 'option:tag': [Function (anonymous)], 'option:dev': [Function (anonymous)] }, _eventsCount: 23, _maxListeners: undefined, commands: [], options: [ [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option] ], parent: Command { _events: [Object: null prototype], _eventsCount: 1, _maxListeners: undefined, commands: [Array], options: [Array], parent: null, _allowUnknownOption: false, _args: [], rawArgs: [Array], _scriptPath: '/home/XXXXXX/.nvm/versions/node/v19.9.0/bin/cypress', _name: 'cypress', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: null, _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: undefined, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '<command> [options]', args: [Array], [Symbol(kCapture)]: false }, _allowUnknownOption: false, _args: [], rawArgs: null, _scriptPath: null, _name: 'run', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: [Function: listener], _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: 0, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '[options]', _description: 'Runs Cypress tests from the CLI without the GUI', _argsDescription: undefined, exit: true, args: [], [Symbol(kCapture)]: false } ] +1ms
  cypress:cli:cli variable-length opts parsed { args: [ '/home/XXXXXX/.nvm/versions/node/v16.20.0/bin/node', '/home/XXXXXX/.nvm/versions/node/v19.9.0/bin/cypress', 'run' ], opts: Command { _events: [Object: null prototype] { 'option:auto-cancel-after-failures': [Function (anonymous)], 'option:browser': [Function (anonymous)], 'option:ci-build-id': [Function (anonymous)], 'option:component': [Function (anonymous)], 'option:config': [Function (anonymous)], 'option:config-file': [Function (anonymous)], 'option:e2e': [Function (anonymous)], 'option:env': [Function (anonymous)], 'option:group': [Function (anonymous)], 'option:key': [Function (anonymous)], 'option:headed': [Function (anonymous)], 'option:headless': [Function (anonymous)], 'option:no-exit': [Function (anonymous)], 'option:parallel': [Function (anonymous)], 'option:port': [Function (anonymous)], 'option:project': [Function (anonymous)], 'option:quiet': [Function (anonymous)], 'option:record': [Function (anonymous)], 'option:reporter': [Function (anonymous)], 'option:reporter-options': [Function (anonymous)], 'option:spec': [Function (anonymous)], 'option:tag': [Function (anonymous)], 'option:dev': [Function (anonymous)] }, _eventsCount: 23, _maxListeners: undefined, commands: [], options: [ [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option] ], parent: Command { _events: [Object: null prototype], _eventsCount: 1, _maxListeners: undefined, commands: [Array], options: [Array], parent: null, _allowUnknownOption: false, _args: [], rawArgs: [Array], _scriptPath: '/home/XXXXXX/.nvm/versions/node/v19.9.0/bin/cypress', _name: 'cypress', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: null, _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: undefined, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '<command> [options]', args: [Array], [Symbol(kCapture)]: false }, _allowUnknownOption: false, _args: [], rawArgs: null, _scriptPath: null, _name: 'run', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: [Function: listener], _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: 0, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '[options]', _description: 'Runs Cypress tests from the CLI without the GUI', _argsDescription: undefined, exit: true, args: [], [Symbol(kCapture)]: false } } +108ms
  cypress:cli parsed cli options {} +112ms
  cypress:cli verifying Cypress app +0ms
  cypress:cli checking environment variables +1ms
  cypress:cli checking if executable exists /home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress +2ms
  cypress:cli Binary is executable? : true +2ms
  cypress:cli binaryDir is  /home/XXXXXX/.cache/Cypress/12.9.0/Cypress +0ms
  cypress:cli Reading binary package.json from: /home/XXXXXX/.cache/Cypress/12.9.0/Cypress/resources/app/package.json +0ms
  cypress:cli Found binary version 12.9.0 installed in: /home/XXXXXX/.cache/Cypress/12.9.0/Cypress +3ms
  cypress:cli could not read binary_state.json file at "/home/XXXXXX/.cache/Cypress/12.9.0/binary_state.json" +3ms
  cypress:cli {} +1ms
  cypress:cli is Verified ? undefined +2ms
  cypress:cli running binary verification check 12.9.0 +0ms
It looks like this is your first time using Cypress: 12.9.0


  cypress:cli clearing out the verified version +6ms
  cypress:cli DISPLAY environment variable is set to localhost:10.0 on Linux
  cypress:cli Assuming this DISPLAY points at working X11 server,
  cypress:cli Cypress will not spawn own Xvfb
  cypress:cli
  cypress:cli NOTE: if the X11 server is NOT working, Cypress will exit without explanation,
  cypress:cli   see https://github.com/cypress-io/cypress/issues/4034
  cypress:cli Solution: Unset the DISPLAY variable and try again:
  cypress:cli   DISPLAY= npx cypress run ... +0ms
  cypress:cli needs Xvfb? false +1ms
  cypress:cli spawning, should retry on display problem? true +0ms
  cypress:cli disabling Electron sandbox +1ms
  cypress:cli running smoke test +0ms
  cypress:cli using Cypress executable /home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress +0ms
  cypress:cli smoke test command: /home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress --no-sandbox --smoke-test --ping=660 +0ms
  cypress:cli smoke test timeout 30000 ms +0ms
  cypress:cli Smoke test failed: Error: Command failed with exit code 127: /home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress --no-sandbox --smoke-test --ping=660
/home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory
    at makeError (/home/XXXXXX/.nvm/versions/node/v19.9.0/lib/node_modules/cypress/node_modules/execa/lib/error.js:59:11)
    at handlePromise (/home/XXXXXX/.nvm/versions/node/v19.9.0/lib/node_modules/cypress/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  shortMessage: 'Command failed with exit code 127: /home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress --no-sandbox --smoke-test --ping=660',
  command: '/home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress --no-sandbox --smoke-test --ping=660',
  exitCode: 127,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: '/home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
} +18ms
  cypress:cli error message: /home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory +1ms
  cypress:cli detecting arch { osPlatform: 'linux', osArch: 'x64' } +37ms

Cypress failed to start.

This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies

Please refer to the error below for more details.

----------

/home/XXXXXX/.cache/Cypress/12.9.0/Cypress/Cypress: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory

----------

Platform: linux-x64 (Raspbian - 15.4)
Cypress Version: 12.9.0
```

### Other

I did:

zypper install mozilla-nss-32bit

but it didn't resolve the issue.

Thx for any advice.

Datapunk

**Comment by mike-plummer** [comment]
@D042657 Can you confirm whether you have the [required prerequisites](https://docs.cypress.io/guides/getting-started/installing-cypress#Linux-Prerequisites) for running Cypress (see the "Linux Prerequisites" section)? It looks like you may have missed installing some of these dependencies

**Comment by D042657** [comment]
Hello Mike, thx for the advice. I was reading the section listing the Linux prerequisites, but could not find a section for SLES, just for Ubuntu and Debian. That's why I searched the web how to make libnss3.so available on SLES and installed the mozilla-nss-32bit package but it didn't resolve the problem. I'll read the prerequisites section again, maybe I missed something.

**Comment by D042657** [comment]
I checked the availability of libnss3.so again, and it seems to be available on my computer.

╰─○ sudo zypper install libnss3.so
......
Loading repository data...
Reading installed packages...
'libnss3.so' not found in package names. Trying capabilities.
'mozilla-nss-32bit' providing 'libnss3.so' is already installed.
Resolving package dependencies...
Nothing to do.

I found it in:

d042657-new:/usr/lib # ls -ltr libnss3.so
-rwxr-xr-x 1 root root 1480860 Feb 10 21:13 libnss3.so

/usr/lib

I added it to the LD_LIBRARY_PATH, then it found the library but complained it's 32bit on 64bit OS version. I'm looking for a package that contains libnss3.so in 64bit version and will try again.
